cms-plugin/src/main/java/com/agiletec/plugins/jacms/aps/system/services/resource/ResourceService.java

```diff
@@ -181,5 +185,9 @@ public PagedMetadata<ComponentUsageEntity> getComponentUsageDetails(String compo
         usageEntries.setBody(sublist);
         return usageEntries;
     }
+    
+    private String extractCorrelationCodeByCode(String code) {
+        return (code.startsWith(CORRELATION_CODE_PARAMETER + "=")) ? code.split("=")[1] : null;
```

Comment:
here I see 2 problems:
- you shouldn't propagate a `null` value, if you really don't have a default return `Optional.empty` and change the return type to `Optional<String>`
- you should reuse code from `ResourcesController.deleteAsset` if possible at least the logic so if not present the cc label you should use the value as code

Reply:
Done, now the method return an Optional and I'd tried to use the same logic of the controller